src/tests/end-to-end/common/browser-factory.ts

```diff
@@ -29,17 +31,7 @@ export async function launchBrowser(extensionOptions: ExtensionOptions): Promise
     const devExtensionPath = `${(global as any).rootDir}/drop/extension/dev/product`;
     const manifestPath = getManifestPath(devExtensionPath);
 
-    let onClose: () => Promise<void>;
-
-    if (extensionOptions.addLocalhostToPermissions) {
-        const originalManifest = await readFile(manifestPath);
-
-        const permissiveManifest = addLocalhostPermissionsToManifest(originalManifest.toString());
-
-        await writeFile(manifestPath, permissiveManifest);
-
-        onClose = async () => await writeFile(manifestPath, originalManifest.toString());
-    }
+    const onClose = await alterManifestWithPermissions(extensionOptions, manifestPath);
```

Comment:
This line feels a little surprising to read (I assumed it was a bug at first based on the function name); maybe more clear would be to move it into a class where there can be separate `setTemporaryManifestPermissions` and `restoreOriginalManifestPermissions` functions?

Reply:
updated.